Fix travis missing requirements.txt

kevinluvian opened the pull request.

kevinluvian committed:
add requirements.txt for travis

kevinluvian committed:
fix requirement configuration in setup.py

kevinluvian commented:
@MihaiTabara Hi, I updated the travis CI configuration
I also added some requirements.txt depending on env in order to make it easier to install dependencies

kevinluvian committed:
remove dummy test, add freezegun lib for mock time

kevinluvian committed:
rename folder tests to test

MihaiTabara commented on test/test_shipitapi.py:
```diff
@@ -19,28 +20,18 @@ def fake(*args, **kwargs):
 
 
+@freeze_time('2018-10-27 00:00:00')
 def test_csrf_token_expired(mocker):
```
This looks good!

MihaiTabara reviewed (approved):
This looks good to me, thanks for addressing the comments from the previous PR!

I see this fixes Travis as well so I'm happy to merge this. If we wanted to be fully consistent with the other repositories, we could be using tox. You can find more about it [here](https://tox.readthedocs.io/en/latest/). It basically handles the management around the virtualenv and enhances the testing automation. 

You can find an example of `tox.ini` [here in beetmoverscript](https://github.com/mozilla-releng/beetmoverscript/blob/master/tox.ini). Most of the fields are intuitive and can be found in the docs, but long story short:

1. you can rename `requirements-base.txt` to `requirements-prod.txt'
2. you can keep the `requirements-dev.txt` for future developers, that looks accurate. 
3. you can move your content from`requirements-test.txt` within the `deps` section of the tox file and remove the `txt` file.  
4. Within the `tox.ini` file you define the `deps` as aforementioned and the `commands` to run. You can keep the `pytest` run from existing `.travis.yml` file but you could also plug-in the coveralls package to also enable code coverage for this project which would be nice to have 👍 Moreover you can also plugin the `flake8` check as well to ensure we're consistent with the PEP8 standards. You can find an example [here](https://github.com/mozilla-releng/beetmoverscript/blob/master/tox.ini#L29).

Once you're all setup with `tox.ini`, you can simply rewrite `travis.yml` to run `tox` instead. You can fine an example [here](https://github.com/mozilla-releng/beetmoverscript/blob/master/.travis.yml#L15).

MihaiTabara commented:
@kevinluvian: I can merge this up since it's green and looks good. If you want to follow-up with `tox` fixes in this PR, let me know and I can hold on from merging. If you want to address that in a follow-up PR, that's also good. If you don't want to do it at all, that's fine as well, no worries. We're likely to make some drastic changes to this repo anyway soon.

kevinluvian commented:
Hi Mihai! I’ll try to learn tox and add it on this pull request. I wonder if you have any way I can contact you (slack/any other communication method) to ask you about this if I have question 😁

MihaiTabara commented:
Sounds good, thanks again for your contribution! I'm `mtabara` on Mozilla's IRC network. I'm usually answering quite fast on github too if you tag my name so feel free to use either. Sorry again for initial delay in answering, been having a busy last week.

MihaiTabara commented:
@kevinluvian For more first-good bugs, feel free to look into https://github.com/mozilla/release-services/issues?q=is%3Aissue+is%3Aopen+label%3Askill%3Agood-first-bug as well. My colleague @garbas can guide you more. Feel free to join #release-services on IRC as well for more questions.

kevinluvian committed:
Add tox for better consistency

- added flake8 for lint
- add mock requirement
- add coverage (without coveralls)
- fix lint for flake8
- remove requirements-test, moved it to tox.ini
- add manifest.in to include requirements-dev and requirements-prod

kevinluvian committed:
add tox dependency

kevinluvian committed:
fix python 3.7

kevinluvian committed:
add coveralls

kevinluvian committed:
update coveragerc

kevinluvian commented:
@MihaiTabara 
Hi, I added tox, flake8 (also fixed some lint issues), and integration to coveralls!

please take a look 🙏

kevinluvian committed:
remove .idea folder

kevinluvian committed:
add endline to some files

MihaiTabara commented on .travis.yml:
```diff
@@ -1,9 +1,23 @@
+      sudo: true
+
+install:
+    - pip install -e ".[dev]"
```
You only need to install `tox` here. Whenever it's run at line 20, it'll install automatically it's own virtual env with all the dependencies that you set up in https://github.com/mozilla-releng/shipitapi/pull/8/files#diff-b91f3d5bd63fcd17221b267e851608e8R15

MihaiTabara commented on setup.py:
```diff
@@ -4,17 +4,26 @@
     author='Mihai Tabara',
     author_email='mtabara@mozilla.com',
+    install_requires=get_requirements('prod'),
+    extras_require={
```
You can remove this. No need for it since any developer can also run `tox` locally and get the same environment of testing as is on Travis CI.

kevinluvian committed:
change travis to only install tox, remove extras_requirements
